pkg/apis/autoscaling/register.go

```diff
@@ -26,11 +26,38 @@ const (
 	//   autoscaling.knative.dev/class: foo
 	// This uses a different domain because unlike the resource, it is user-facing.
 	ClassAnnotationKey = GroupName + "/class"
-
+	// KPA is Knative Horizontal Pod Autoscaler
+	KPA = "kpa"
+	// HPA is Kubernetes Horizontal Pod Autoscaler
+	HPA = "hpa"
+
+	// MinScaleAnnotationKey is the annotation to specify the minimum number of Pods
+	// the PodAutoscaler should provision. For example,
+	//   autoscaling.knative.dev/minScale: "1"
 	MinScaleAnnotationKey = GroupName + "/minScale"
+	// MinScaleAnnotationKey is the annotation to specify the maximum number of Pods
+	// the PodAutoscaler should provision. For example,
+	//   autoscaling.knative.dev/maxScale: "10"
 	MaxScaleAnnotationKey = GroupName + "/maxScale"
 
+	// MetricAnnotationKey is the annotation to specify what metric the PodAutoscaler
+	// should be scaled on. For example,
+	//   autoscaling.knative.dev/metric: cpu
+	MetricAnnotationKey = GroupName + "/metric"
+	// Concurrency is the number of requests in-flight at any given time.
+	Concurrency = "concurrency"
+	// OPS is the number of requests (operations) arriving per second.
+	OPS = "ops"
+	// CPU is the amount of the requested cpu actually being consumed by the Pod.
+	CPU = "cpu"
+
+	// TargetAnnotationKey is the annotation to specify what metric value the
+	// PodAutoscaler should attempt to maintain. For example,
+	//   autoscaling.knative.dev/metric: cpu
+	//   autoscaling.knative.dev/target: 75   # target 75% cpu utilizationa
+	TargetAnnotationKey = GroupName + "/target"
```

Comment:
Will we eventually use this in the KPA as well, forwarding revision's `containerConcurrency` to the target annotation on KPA?

Reply:
Yes :)

The Revision controller will still only copy annotations from Revision to PA.  The hpa-class PA controller will be sensitive to the `/target` annotation.  It will look at the `Spec.ContainerCurrency` and `Annotations[autoscaling.knative.dev]` and determine the right effective target.  Seems okay to me, no?

Reply:
What I was saying "No" to was having the Revision controller put containerConcurrency into the annotation instead of a field.  I wasn't suggesting we need any change here, just answering @markusthoemmes comment ;-)